examples/gke-daemonset-raid-disks.yaml

```diff
@@ -0,0 +1,54 @@
+# 1. Run gcloud CLI with --local-nvme-ssd-block option to create a node pool with
+# raw block Local SSDs attached.
+# 2. Deploy this DaemonSet. It will set RAID0 array on all raw block Local SSD
+# disks on each node.
+# 3. Use the gke-nvme-ssd-block.yaml to create PV and StorageClass
+# 4. Use gke-pvc-nvme-ssd-block.yaml and gke-pod-nvme-ssd-block.yaml to create
+# pvc and pod
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: gke-raid-disks
+  namespace: default
+  labels:
+    k8s-app: gke-raid-disks
+spec:
+  selector:
+    matchLabels:
+      name: gke-raid-disks
+  template:
+    metadata:
+      labels:
+        name: gke-raid-disks
+    spec:
+      nodeSelector:
+        cloud.google.com/gke-local-nvme-ssd: "true"
+      hostPID: true
+      containers:
+      - name: startup-script
+        image: gcr.io/google-containers/startup-script:v1
+        securityContext:
+          privileged: true
+        env:
+        - name: STARTUP_SCRIPT
+          value: |
+            devices=()
+            for ssd in /dev/disk/by-id/google-local-ssd-block*; do
+              if [ -e "${ssd}" ]; then
+                devices+=("${ssd}")
+              fi
+            done
+            if [ "${#devices[@]}" -eq 0 ]; then
+              echo "No Local NVMe SSD disks found."
+              return
+            fi
+
+            if [ "${#devices[@]}" -ne 1 ]; then
```

Comment:
Why is 1 skipped?

Reply:
I thought one disk is a special case, but I agree we should make it consistent cross all cases. I'll update it. Another issue I notice is when I reboot the instance /dev/md/0 become /dev/md/0_0 which is not done by my script. And local static provisioner create a new PV on /dev/md/0_0 with the old PV on /dev/md/0 still exist. Is this a expected behavior? Can local static provisioner cleanup PV when the underlying device changed?

Reply:
Discussed with @msau42 offline, RAID rename device at instance reboot is a known issue https://serverfault.com/questions/494569/mdadm-raid-device-name-changed-on-reboot Add the filesystem formatting to the script to avoid failure